tools/tempfs/tempfs.go

```diff
@@ -0,0 +1,36 @@
+package tempfs
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/google/oss-rebuild/pkg/rebuild/rebuild"
+	"github.com/pkg/errors"
+)
+
+// /tmp/oss-rebuild/
+// 		assets/
+//		firestore/
+
+const (
+	TempRoot = "/tmp/oss-rebuild/"
+)
+
+func FirestoreFS() billy.Filesystem {
```

Comment:
Depending on the library name, wdyt of `localfiles.Firestore()`?

Reply:
Is it firestore tho? The abstraction is more generally 'metadata' with firestore being how we currently implement it. I get the logic behind calling it firestore but something more general like temp.MetadataFS would probably be more appealing to me. My level of caring is probably 2/10 tho. Easy to refactor after.

Reply:
The filepath we're providing access for is /tmp/oss-rebuild/firestore

If we want to change the implementation in the future, I think we'd want to change the directory we're using to store the files in, and consequently we'd want to use a different accessor, which makes me think it's reasonable to use Firestore as the name for this.

Reply:
I don't agree. The current interface doesn't really need to have any association with firestore, just the remote client. Naming the path and the overall concept as "firestore" means we couple that usage to a (rather arbitrary) storage choice. It's not important, though, so feel free to resolve.

Reply:
Discussed offline, I think the point you were making applied more to the library rather than this specific function. I renamed the library instead of "firestore" it's now "rundex".